utils.lua

```diff
@@ -144,4 +144,18 @@ end
 
 function Hardcore_IsNumber(val)
 	return type(val) == "number"
+end
+
+function Hardcore_FilterUnique(tbl)
```

Comment:
This is pretty interesting and cool! It would have been nice if lua had built in sets